### PR TITLE
Provide a view to get protocol interfaces

### PR DESCRIPTION
**Pull request by jonc125**
This is the first part of the work on #65, but can sensibly be reviewed as a standalone chunk.

It builds on #109, hence the base branch selected.

I still need to check that it's running the number of DB queries I expect...

Child of #68.

### CODE REVIEW

**Comment by jonc125** [comment]
Having fixed the debug logging on a dev deploy (oops!) I've now checked that fetching http://localhost:8088/entities/protocols/get_interfaces does indeed result in the correct output and only 2 DB queries being made, so think this is ready for review.

File: weblab/repocache/populate.py [hunk 1]
```diff
@@ -13,25 +11,26 @@ def populate_entity_cache(entity):
...
-        version = CachedEntityVersion.objects.create(
+        version = CachedEntityVersion.objects.get_or_create(
             entity=cached,
             sha=commit.hexsha,
-            timestamp=commit.committed_at,
-            visibility=visibility or entity.DEFAULT_VISIBILITY
-        )
+            defaults={
+                'timestamp': commit.committed_at,
+                'visibility': visibility or entity.DEFAULT_VISIBILITY,
```
Comment (helenst) [inline weblab/repocache/populate.py]: This doesn't seem like it'll update the cache if the visibility of a version changes?

Comment (jonc125) [reply weblab/repocache/populate.py]: Good spot, yes. I'll add a test case for that!

**Comment by jonc125** [comment]
Fixed the issue you spotted and the merge conflict; now ready for re-review @helenst .

**Comment by helenst** [comment]
Looks good to me :+1:

**Pull request merged**